server/notification-providers/threema.js

```diff
@@ -0,0 +1,83 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+
+class Threema extends NotificationProvider {
+    name = "threema";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        const url = "https://msgapi.threema.ch/send_simple";
+
+        const config = {
+            headers: {
+                "Accept": "*/*",
+                "Content-Type": "application/x-www-form-urlencoded; charset=utf-8"
+            }
+        };
+
+        const data = {
+            from: notification.threemaSenderIdentity,
+            secret: notification.threemaSecret,
+            text: msg
+        };
+
+        switch (notification.threemaRecipientType) {
+            case "identity":
+                data.to = notification.threemaRecipient;
+                break;
+            case "phone":
+                data.phone = notification.threemaRecipient;
+                break;
+            case "email":
+                data.email = notification.threemaRecipient;
+                break;
+            default:
+                throw new Error(`Unsupported recipient type: ${notification.threemaRecipientType}`);
+        }
+
+        try {
+            await axios.post(url, new URLSearchParams(data), config);
+            return "Threema notification sent successfully.";
+        } catch (error) {
+            this.handleApiError(error);
+            this.throwGeneralAxiosError(error);
+        }
+    }
+
+    /**
+     * Handle Threema API errors
+     * @param {any} error The error to handle
+     * @returns {void}
+     */
+    handleApiError(error) {
+        if (error.response) {
+            const status = error.response.status;
+            switch (status) {
+                case 400:
+                    error.message = "Invalid recipient identity or account not set up for basic mode (400).";
+                    break;
+                case 401:
+                    error.message = "Incorrect API identity or secret (401).";
+                    break;
+                case 402:
+                    error.message = "No credits remaining (402).";
+                    break;
+                case 404:
+                    error.message = "Recipient not found (404).";
+                    break;
+                case 413:
+                    error.message = "Message is too long (413).";
+                    break;
+                case 500:
+                    error.message = "Temporary internal server error (500).";
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
```

Comment:
I somewhat consider this kind of reaching into the error message and changing it an antipattern.
Reasoning about this is quite difficult.
Please semi-inline the parts instead and handle just the status code switching in this function

```suggestion
    handleApiError(error) {
        if (!error.response) {
            return error.message;
        }
        switch (error.response.status) {
            case 400:
                return "Invalid recipient identity or account not set up for basic mode (400).";
            case 401:
                return "Incorrect API identity or secret (401).";
            case 402:
                return "No credits remaining (402).";
            case 404:
                return "Recipient not found (404).";
            case 413:
                return "Message is too long (413).";
            case 500:
                return "Temporary internal server error (500).";
            default:
                return error.message;
        }
    }
```

Reply:
Makes sense. With the suggested approach, the message would then be directly passed to `throwGeneralAxiosError` like this?
```
const errorMessage = this.handleApiError(error);
this.throwGeneralAxiosError(errorMessage);
```